testar/src/org/fruit/monkey/Main.java

```diff
@@ -105,7 +106,10 @@ public static String getTestSettingsFile() {
 	 */
 	public static void main(String[] args) throws IOException {
 
-		isValidJavaEnvironment();
+		if(! isValidJavaEnvironment()) {
+			System.out.println("No valid java environment detected");
+			System.exit(1);
```

Comment:
Thanks for contributing! 

We tested the code and noticed that this code is not executed at all with Java 1.7 or older (JVM throws an error when trying to start the main method). Therefore, the check does not work for older Java versions. I think this could be fixed by checking Java version in testar.bat instead of in the TESTAR code.

Instead, it will only check if a newer Java version (1.8<) is used. We have not tested much with newer Java versions but with a quick test, for example Java 12 seems to work. 

Therefore we should not System.exit(1) but instead we should warn that TESTAR has not been tested properly with other than Java 1.8 - use at your own risk.